src/panels/lovelace/editor/config-elements/hui-weather-forecast-card-editor.ts

```diff
@@ -57,14 +57,15 @@ export class HuiWeatherForecastCardEditor
 
     if (
       /* cannot show forecast in case it is unavailable on the entity */
-      (config.show_forecast === true && this._hasForecast === false) ||
+      (config.show_forecast !== false && this._hasForecast === false) ||
       /* cannot hide both weather and forecast, need one of them */
       (config.show_current === false && config.show_forecast === false)
     ) {
       /* reset to sane default, show weather, but hide forecast */
       fireEvent(this, "config-changed", {
         config: { ...config, show_current: true, show_forecast: false },
       });
+      return;
```

Comment:
If we don't return, we might fire a second fireEvent in the next code section, which would overwrite what we fired here and potentially spin an infinite loop.